pyperformance/data-files/benchmarks/bm_deepcopy/run_benchmark.py

```diff
@@ -0,0 +1,42 @@
+"""
+Benchmark to measure performance of the python builtin method copy.deepcopy
+
+Performance is tested on a nested dictionary and a dataclass
+
+Author: Pieter Eendebak
+
+"""
+import copy
+import pyperf
+from dataclasses import dataclass
+
+@dataclass
+class A:
+    string : str
+    lst : list
+    boolean : bool
+    
+
+def benchmark(n):
+    a={'list': [1,2,3,43], 't': (1,2,3), 'str': 'hello', 'subdict': {'a': True}}
+    dc=A('hello', [1,2,3], True)
```

Comment:
If the objective of this micro benchmark is to monitor for `copy.deepcopy()` performance regressions, then we need to be sure the benchmark has sufficient code coverage.  (It may even be worth running coverage.py on it.)  At the least consider deepcopying other supported objects, including objects that implement different parts of the copy protocol (`__reduce__()`, `__setstate__()`, etc.).

Reply:
I added 2 more benchmarks for the `__reduce__/__setstate__` and `memo` functionality in `deepcopy`. @ericsnowcurrently